Bottom: f9ef4835cd7e2114c1ea350e11a451359964c503
Top:    9d0416103b45249fd4d7c7c3aa9acfd40e66434a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-28 00:43:48 +0100

Refresh of drop-in-pcb-gl-code-various-mess

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 5541056..b2a77e0 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -611,12 +611,18 @@ ghid_invalidate_lr (int left, int right, int top, int bottom)
 void
 ghid_invalidate_all ()
 {
-  if (ghidgui->need_restore_crosshair)
-    RestoreCrosshair ();
-  ghidgui->need_restore_crosshair = FALSE;
   ghid_draw_area_update (gport, NULL);
 }
 
+void
+ghid_notify_crosshair_change (bool changes_complete)
+{
+  /* FIXME: We could just invalidate the bounds of the crosshair attached objects? */
+  /* FIXME: Seems we're getting prodded before the GUI is up! */
+  if (gport->drawing_area)
+    ghid_invalidate_all ();
+}
+
 static void
 draw_right_cross (gint x, gint y)
 {
